app/src/main/java/slowscript/warpinator/TransfersAdapter.java

```diff
@@ -86,6 +83,12 @@ public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
                         .setTitle("Errors during transfer:")
                         .setMessage(Joiner.on("\n").join(t.errors))
                         .show();
+            } else if (t.getStatus() == Transfer.Status.TRANSFERRING) {
+                long now = System.currentTimeMillis();
+                float avgSpeed = t.bytesTransferred / ((now - t.actualStartTime) / 1000f);
+                int secondsRemaining = (int)((t.totalSize - t.bytesTransferred) / avgSpeed);
+                String remainingTime = formatTime(secondsRemaining) + " " + activity.getString(R.string.remaining);
+                Toast.makeText(activity, remainingTime, Toast.LENGTH_SHORT).show();
```

Comment:
Remaining time is useful information. I think this should be always visible, not just in a toast after tapping the card, which the user doesn't intuitively know is possible.

Reply:
well said. something definitely needs to be moved out of there though. or we should expand the card(this might look weird on tablets)?

Reply:
Idk, the entire string used to fit well on every device I tried. My phone is 5 inches which is the smallest that is available nowadays. I will have a look at this when I have time.